### PR TITLE
Iox #33 disable warnings on windows

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer

The compiler warnings for MSVC are disabled. I also added a todo to #33 to re-enable them when Windows is fully supported.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Related to #33

### CODE REVIEW

**Review by dkroenke — APPROVED** [review]

File: iceoryx_binding_c/test/CMakeLists.txt [hunk 1]
```diff
@@ -57,5 +58,5 @@ endif()
 add_executable(${PROJECT_PREFIX}_moduletests ${MODULETESTS_SRC})
 set_property(TARGET ${PROJECT_PREFIX}_moduletests PROPERTY CXX_STANDARD ${ICEORYX_CXX_STANDARD})
 target_include_directories(${PROJECT_PREFIX}_moduletests PRIVATE .)
-target_compile_options(${PROJECT_PREFIX}_moduletests PUBLIC -fPIC PRIVATE ${TEST_CXX_FLAGS})
+target_compile_options(${PROJECT_PREFIX}_moduletests PRIVATE ${TEST_CXX_FLAGS})
```
Comment (mossmaurice) [inline iceoryx_binding_c/test/CMakeLists.txt:61]: We don't need a ``POSITION_INDEPENDENT_CODE ON` here as fPIC is only for lib builds?

File: iceoryx_utils/test/CMakeLists.txt [hunk 2]
```diff
@@ -45,7 +45,7 @@ elseif(QNX)
 elseif(APPLE)
     set(TEST_CXX_FLAGS PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS} -Wno-pedantic -Wno-conversion)
 elseif(WIN32)
-    set(TEST_CXX_FLAGS PRIVATE /bigobj)
+    set(TEST_CXX_FLAGS PRIVATE ${ICEORYX_WARNINGS} /bigobj)
```
Comment (mossmaurice) [inline iceoryx_utils/test/CMakeLists.txt:48]: Please update the copyright header

File: iceoryx_utils/test/stresstests/benchmark_optional_and_expected/CMakeLists.txt [hunk 3]
```diff
@@ -33,7 +33,18 @@ target_link_libraries(iox-bm-optional-and-expected
     iceoryx_utils::iceoryx_utils
     Threads::Threads
 )
-set_target_properties(iox-bm-optional-and-expected PROPERTIES 
+
```
Comment (mossmaurice) [inline iceoryx_utils/test/stresstests/benchmark_optional_and_expected/CMakeLists.txt:36]: Please update the copyright header

File: iceoryx_binding_c/test/CMakeLists.txt (hunk 1 above)
Comment (elBoberido) [reply iceoryx_binding_c/test/CMakeLists.txt:61]: I think it's a leftover from a previous cleanup. It's not used anywhere else.

**Review by dkroenke — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]

**Pull request merged**